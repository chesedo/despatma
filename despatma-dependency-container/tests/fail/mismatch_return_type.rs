struct Config {
    port: u32,
}

struct Unit;

struct Service;

impl Service {
    fn new(port: u32, _unit: Unit) -> Self {
        println!("Service started on port {}", port);
        Self
    }
}

#[despatma_dependency_container::dependency_container]
impl DependencyContainer {
    fn config(&self) -> Config {
        Config { port: 8080 }
    }

    fn unit(&self) -> () {
        ()
    }

    fn service(&self, config: u32, unit: Unit) -> Service {
        Service::new(config, unit)
    }
}

fn main() {
    let container = DependencyContainer::new();
    let _service = container.service();
}
