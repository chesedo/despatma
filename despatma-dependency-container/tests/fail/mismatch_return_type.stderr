error[E0308]: arguments to this function are incorrect
  --> tests/fail/mismatch_return_type.rs:27:9
   |
27 |         Service::new(config, unit)
   |         ^^^^^^^^^^^^ ------  ---- expected `Unit`, found `()`
   |                      |
   |                      expected `u32`, found `Config`
   |
note: associated function defined here
  --> tests/fail/mismatch_return_type.rs:10:8
   |
10 |     fn new(port: u32, _unit: Unit) -> Self {
   |        ^^^ ---------  -----------
